src/DynamoManipulation/Gizmo.cs

```diff
@@ -171,10 +171,17 @@ private void Redraw()
 
         private void OnViewCameraChanged(object o, RoutedEventArgs routedEventArgs)
         {
-            cameraPosition = routedEventArgs.Source as Point3D?;
-
-            // Redraw Gizmos
-            Redraw();
+            var newCameraPosition = routedEventArgs.Source as Point3D?;
+            // A change of behavior in helix when upgrading from v2015 is that the CameraChanged event gets fired
+            // several times, once for each individual property of Camera that changed.
+            // In order to avoid rendering the Gizmo several times, which leads to choppy camera rotation, we limit the
+            // rendering of the Gizmo to only be done when the position of the camera changed.
+            if (cameraPosition != newCameraPosition)
```

Comment:
hmm 
https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.media3d.point3d.op_equality?view=netframework-4.8#System_Windows_Media_Media3D_Point3D_op_Equality_System_Windows_Media_Media3D_Point3D_System_Windows_Media_Media3D_Point3D_

do you think it's worth to do an epsilon tolerant equals?

Reply:
this is what I'm referring to:
```
Two Point3D structures are equal if the values of their X, Y , and Z properties are the same.

Because Double values can lose precision when arithmetic operations are performed on them, a comparison between two Point3D instances that are logically equal might fail.

```

Reply:
Due to the nature of the issue, there shouldn't be a need to be epsilon tolerant. Basically what's happening is:
1. Camera.LookDirection is set -> OnViewCameraChanged (ignored as Position is same)
2. Camera.Position is set -> OnViewCameraChanged (not ignored as Position is different)
3. Camera.UpDirection is set -> OnViewCameraChanged (ignored as Position is same)

Reply:
Would the default **Equals** sufficient? https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.media3d.point3d.equals?view=netframework-4.8#System_Windows_Media_Media3D_Point3D_Equals_System_Object_

Reply:
@QilongTang Yes, we are ignoring case 1 and 3 in my post above by using (X, Y, Z) equality. The `==` operator is overloaded to behave exactly as `Equals`.

Reply:
cool, thanks for the explanation @mmisol